node/src/node/slave/machine.cpp

```diff
@@ -237,6 +248,8 @@ auto machine_t::inject(load_t& load, channel_handler handler) -> std::uint64_t {
         });
     }
 
+    metrics_data.update_concurrency(current);
```

Comment:
Definitely it make the wrapped object thread-safe, but depending on the execution context sometimes it's not necessary.